internal/services/cdn/frontdoorsecurityparams/cdn_frontdoor_security_params.go

```diff
@@ -0,0 +1,146 @@
+package cdnfrontdoorsecurityparams
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/cdn/mgmt/2021-06-01/cdn"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CdnFrontdoorSecurityParameters struct {
+	TypeName   cdn.Type
+	ConfigName string
+}
+
+type CdnFrontdoorSecurityMappings struct {
+	Firewall CdnFrontdoorSecurityParameters
+}
+
+func InitializeCdnFrontdoorSecurityMappings() *CdnFrontdoorSecurityMappings {
+	m := new(CdnFrontdoorSecurityMappings)
+
+	m.Firewall = CdnFrontdoorSecurityParameters{
+		TypeName:   cdn.TypeWebApplicationFirewall,
+		ConfigName: "firewall",
+	}
+
+	return m
+}
+
+func ExpandCdnFrontdoorFirewallPolicyParameters(input []interface{}, isStandardSku bool) (cdn.SecurityPolicyWebApplicationFirewallParameters, error) {
```

Comment:
should be a pointer:

```suggestion
func ExpandCdnFrontdoorFirewallPolicyParameters(input []interface{}, isStandardSku bool) (*cdn.SecurityPolicyWebApplicationFirewallParameters, error) {
```

Reply:
Fixed.